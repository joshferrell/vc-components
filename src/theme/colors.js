const primary = {
	main: '#202d4a',
	secondary: '#ffffff',
	tertiary: '#afbfff'
};

const secondary = {
	main: '#f7f7f7',
	secondary: '#2b2b2b',
	tertiary: '#5677fc'
};

const tertiary = {
	main: '#009688',
	secondary: '#ffffff',
	tertiary: '#ffffff'
};

const highlight = {
	red: '#e51c23',
	green: '#259b24',
	blue: '#5677fc',
	yellow: '#ffeb3b'
};

const colors = {
	primary,
	secondary,
	tertiary,
	highlight
};

export default colors;
