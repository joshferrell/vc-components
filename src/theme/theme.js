import React from 'react';
import PropTypes from 'prop-types';
import styled, { ThemeProvider } from 'styled-components';
import { themeGet } from 'styled-system';
import { systemFonts } from 'utils';
import { lighten } from 'polished';
import colors from './colors';

const theme = {
	fonts: {
		0: `Poppins, ${systemFonts}`
	},
	fontSizes: [0.819, 1.092, 1.455, 1.94, 2.585, 3.446, 4.594],
	shadows: [
		'none',
		'rgba(50, 50, 93, 0.11) 0px 4px 6px 0px, rgba(0, 0, 0, 0.08) 0px 1px 3px 0px',
		'0 0 16px rgba(0, 0, 0, 0.1)',
		'0 10px 20px rgba(0, 0, 0, 0.1), 0 3px 6px rgba(0, 0, 0, 0.1)'
	],
	space: [0, 4, 8, 16, 24, 32, 64],
	lineHeights: [1, 1.125, 1.25, 1.5, 1.75],
	colors
};

const BaseFont = styled.div`
	font-family: ${themeGet('fonts.0')};
	text-rendering: optimizeLegibility;
	font-feature-settings: none;
	font-variant-ligatures: none;
	-moz-osx-font-smoothing: grayscale;
	-webkit-font-smoothing: antialiased;

	button,
	input,
	optgroup,
	select,
	textarea {
		font-family: ${themeGet('fonts.0')};
	}

	::selection {
		background: rgba(0, 122, 255, .4);
	}
`;

const VoiceTheme = ({ children }) => (
	<ThemeProvider theme={theme}>
		<BaseFont>
			{children}
		</BaseFont>
	</ThemeProvider>
);

VoiceTheme.propTypes = {
	children: PropTypes.node.isRequired
};

export default VoiceTheme;
