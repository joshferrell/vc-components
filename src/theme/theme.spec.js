import { theme } from './theme';

describe('voice computer theme', () => {
	it('should have a theme with colors and fonts', () => {
		expect(theme).toMatchSnapshot();
	});
});
