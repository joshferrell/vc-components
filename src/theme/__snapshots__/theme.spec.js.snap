// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`voice computer theme should have a theme with colors and fonts 1`] = `
Object {
  "colors": Object {
    "highlight": Object {
      "blue": "#5677fc",
      "green": "#259b24",
      "red": "#e51c23",
      "yellow": "#ffeb3b",
    },
    "neutral": Array [
      "#fafafa",
      "#eeeff1",
      "#d2d5db",
      "#a6abb7",
      "#4d586f",
      "#202D4A",
      "#1a2a4d",
    ],
    "primary": Object {
      "main": "#202d4a",
      "secondary": "#ffffff",
      "tertiary": "#afbfff",
    },
    "secondary": Object {
      "main": "#eeeff1",
      "secondary": "#202D4A",
      "tertiary": "#5677fc",
    },
    "tertiary": Object {
      "main": "#009688",
      "secondary": "#ffffff",
      "tertiary": "#ffffff",
    },
  },
  "fontSizes": Array [
    0.819,
    1.092,
    1.455,
    1.94,
    2.585,
    3.446,
    4.594,
  ],
  "fonts": Object {
    "0": "Poppins, -apple-system, BlinkMacSystemFont, \\"Segoe UI\\", Roboto, Helvetica, Arial, sans-serif, \\"Apple Color Emoji\\", \\"Segoe UI Emoji\\", \\"Segoe UI Symbol\\"",
  },
  "lineHeights": Array [
    1,
    1.125,
    1.25,
    1.5,
    1.75,
  ],
  "shadows": Array [
    "none",
    "rgba(50, 50, 93, 0.11) 0px 4px 6px 0px, rgba(0, 0, 0, 0.08) 0px 1px 3px 0px",
    "0 0 16px rgba(0, 0, 0, 0.1)",
    "0 10px 20px rgba(0, 0, 0, 0.1), 0 3px 6px rgba(0, 0, 0, 0.1)",
  ],
  "space": Array [
    0,
    4,
    8,
    16,
    24,
    32,
    64,
  ],
}
`;
